src/app/api/routes/accesses.py

```diff
@@ -1,28 +1,31 @@
 from typing import List, Dict, Any
-from fastapi import APIRouter, Path, HTTPException
+from fastapi import APIRouter, Path, HTTPException, Security
 from app.api import crud, security
 from app.db import accesses
 from app.api.schemas import AccessBase, AccessRead, AccessAuth, AccessCreation, Cred, CredHash, Login
+from app.api.deps import get_current_access
 
 
 router = APIRouter()
 
 
-async def update_access_login(login: str, access_id: int):
+async def update_access_login(login: str, access_id: int, _=Security(get_current_access, scopes=["admin"])):
```

Comment:
Isn't this dangerous?
Considering update_access_login can be used by other routers I mean. We could update the scope of the routes that uses this no ? (so that we only have scope constraints on routes and not functions that can be used within a route, which could generate cumbersome errors if we accept the request but reject the scope midway)

Reply:
You are right the dependency at function level (not route) seems to have no impact: the route PUT device/{device_id}/ calls `update_access_login` without requiring admin scope. The logic of rejection on unauthorised request must be implemented at route level (devices and users) as mentioned in https://github.com/pyronear/pyro-api/issues/45

Reply:
Mind updating the PR accordingly? :pray:

Reply:
Any update? :grin: